tools/c7n_oci/c7n_oci/provider.py

```diff
@@ -19,15 +21,90 @@ class OCI(Provider):
+            options["oci_config"] = self._get_oci_profile_session(options).get_config()
         return options
 
+    def _get_subscribed_regions(self, options):
+        client = self._get_oci_profile_session(options).client("oci.identity.IdentityClient")
+        subscribed_regions = client.list_region_subscriptions(
+            tenancy_id=options["oci_config"]["tenancy"]
+        ).data
+        regions = [region.region_name for region in subscribed_regions if region.status == "READY"]
+        return regions
+
+    @staticmethod
+    def _is_subscribed_region(subscribed_regions, region):
+        if region in subscribed_regions:
+            return True
+
+        log.info(
+            (
+                "User's tenancy is not subscribed to the %s region. "
+                "So skipping this region for the policy execution."
+            ),
+            region,
+        )
+        return False
+
+    def _get_oci_profile_session(self, options):
+        if self._oci_profile_session is None:
+            self._oci_profile_session = self.get_session_factory(options)()
+        return self._oci_profile_session
+
+    @staticmethod
+    def _join_output(output_dir, suffix):
+        if "{region}" in output_dir:
+            return output_dir.rstrip("/")
+        if output_dir.endswith("://"):
+            return output_dir + suffix
+        output_url_parts = urlparse.urlparse(output_dir)
+        # for output urls, the end of the url may be a
+        # query string. make sure we add a suffix to
+        # the path component.
+        output_url_parts = output_url_parts._replace(
+            path=output_url_parts.path.rstrip("/") + "/%s" % suffix
+        )
+        return urlparse.urlunparse(output_url_parts)
+
     def initialize_policies(self, policy_collection, options):
-        return policy_collection
+        from c7n.policy import Policy, PolicyCollection
+
+        if not options.regions:
+            return policy_collection
+        policies = []
+        subscribed_regions = self._get_subscribed_regions(options)
+        if "all" in options.regions:
+            regions = subscribed_regions
+        else:
+            regions = [
+                r for r in options.regions if self._is_subscribed_region(subscribed_regions, r)
+            ]
+        for p in policy_collection:
+            for region in regions:
+                options_copy = copy.copy(options)
+                options_copy.region = str(region)
+                if (
+                    len(options.regions) > 1
+                    or "all" in options.regions
+                    and getattr(options, "output_dir", None)
+                ):
+                    options_copy.output_dir = self._join_output(options.output_dir, region)
+                policies.append(
+                    Policy(
+                        p.data,
+                        options_copy,
+                        session_factory=policy_collection.session_factory(),
```

Comment:
instantiating the session_factory should create a session, not another factory?

Reply:
Done..Passed the OCI session factory and it's instantiation will create the session.

Reply:
I also want to highlight here that we will have one session per region and that session is used for the execution of the policy.